src/htmx.js

```diff
@@ -1064,11 +1064,15 @@ return (function () {
 
             socket.onclose = function (e) {
                 if ([1006, 1012, 1013].includes(e.code)) {  // Abnormal Closure/Service Restart/Try Again Later
+                    var delay = getWebSocketReconnectDelay(retryCount);
                     setTimeout(function() {
-                        processWebSocketSource(elt, wssSource);  // creates a websocket with a new timeout
-                    }, htmx.config.wsReconnectInterval);
+                        processWebSocketSource(elt, wssSource, retryCount+1);  // creates a websocket with a new timeout
+                    }, delay);
                 }
             };
+            socket.onopen = function (e) {
+                retryCount = 0;
```

Comment:
Should retryCount be defined before `socket.onclose =...` so that it's part of the closure?  I can't tell for sure but it looks like it's a global right now...

Reply:
Oh damn! My fault. I reverted another intermediate change and removed the parameter from the respective function. Should be a local variable indeed.